### PR TITLE
API-2220: Validate scopes for user-restricted endpoints (initial commit)

### PR DESCRIPTION
**Pull request by gokyo**

### CODE REVIEW

File: app/uk/gov/hmrc/controller/ProxyController.scala [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+        case e: InvalidScope => Forbidden(Json.toJson(e))
+        case e => {
+          Logger.error("Unexpected error", e);
+          InternalServerError(Json.toJson(ServerError()))
+        }
+      }
+  }
+}
+
+object ProxyController extends ProxyController {
```
Comment (edgarjimenez) [inline app/uk/gov/hmrc/controller/ProxyController.scala]: as per our earlier conversation, we are planning to use injection, as it is Play's way forward. This unfortunately means there will be some refactoring to be done for this and other controllers you've created.

File: app/uk/gov/hmrc/controller/ProxyController.scala [hunk 2]
```diff
@@ -0,0 +1,35 @@
+package controllers
```
Comment (edgarjimenez) [inline app/uk/gov/hmrc/controller/ProxyController.scala]: the packaging structure has been changed to uk.gov.hmrc.apigateway.controller...

File: app/uk/gov/hmrc/service/ProxyService.scala [hunk 3]
```diff
@@ -0,0 +1,14 @@
+package services
+
+import play.api.mvc._
+import scala.concurrent.Future
+
+trait ProxyService {
+
+  def proxyRequest(request: Request[AnyContent]): Future[Result] = {
```
Comment (edgarjimenez) [inline app/uk/gov/hmrc/service/ProxyService.scala]: I think you'll have to rebase against master, as a lot of this components have been implemented.

**Comment by gokyo** [comment]
Ignore this branch.
I will work on a fresh one.

**Comment by gokyo** [comment]
Closing PR

**Pull request closed without merging**